app/classes/checksum.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+class Checksum
```

Comment:
I'm not sure `classes` is really the right way to go from a Rails perspective. It would almost make more sense moving this into a service or a concern rather than add an entirely new concept.